### PR TITLE
reboot external rhcs nodes

### PR DESCRIPTION
**Pull request by shylesh**
A reboot function for external rhcs nodes
Signed-off-by: Shylesh Kumar Mohan <shmohan@redhat.com>

### CODE REVIEW

**Comment by shylesh** [comment]
20:36:20 - MainThread - ocs_ci.ocs.external_ceph - INFO - Running command reboot on x.x.x.x
20:36:20 - MainThread - ocs_ci.ocs.external_ceph - INFO - long running command --
20:36:21 - Thread-8 - paramiko.transport - INFO - Connected (version 2.0, client OpenSSH_7.4)
20:36:21 - Thread-8 - paramiko.transport - INFO - Authentication (publickey) failed.
20:36:21 - Thread-8 - paramiko.transport - INFO - Authentication (password) successful!
20:36:22 - MainThread - ocs_ci.ocs.external_ceph - INFO - 
20:36:22 - MainThread - ocs_ci.ocs.external_ceph - INFO - Connecting ceph-fedora-1602715266115-node1-monmgrinstaller / x.x.x.x
**20:36:22 - MainThread - ocs_ci.ocs.external_ceph - WARNING - Connection outage: 
[Errno None] Unable to connect to port 22 on x.x.x.x**
20:36:35 - Thread-10 - paramiko.transport - INFO - Connected (version 2.0, client OpenSSH_7.4)
20:36:36 - Thread-10 - paramiko.transport - INFO - Authentication (publickey) failed.
20:36:37 - Thread-10 - paramiko.transport - INFO - Authentication (password) successful!
20:36:38 - MainThread - ocs_ci.ocs.external_ceph - INFO - Running command ls / ; uptime ; date on x.x.x.x
20:36:38 - Thread-12 - paramiko.transport - INFO - Connected (version 2.0, client OpenSSH_7.4)
20:36:39 - Thread-12 - paramiko.transport - INFO - Authentication (publickey) failed.
20:36:39 - Thread-12 - paramiko.transport - INFO - Authentication (password) successful!
20:36:40 - MainThread - ocs_ci.ocs.external_ceph - INFO - Command completed successfully
20:36:40 - MainThread - ocs_ci.ocs.external_ceph - INFO - Running command hostname on x.x.x.x
20:36:40 - MainThread - ocs_ci.ocs.external_ceph - INFO - Command completed successfully
20:36:40 - MainThread - ocs_ci.ocs.external_ceph - INFO - hostname and shortname set to ceph-fedora-1602715266115-node1-monmgrinstaller and ceph-fedora-1602715266115-node1-monmgrinstaller
20:36:40 - MainThread - ocs_ci.ocs.external_ceph - INFO - Running command /sbin/ifconfig eth0 | grep 'inet ' | awk '{ print $2}' on x.x.x.x
20:36:41 - MainThread - ocs_ci.ocs.external_ceph - INFO - Command completed successfully
20:36:41 - MainThread - ocs_ci.ocs.external_ceph - INFO - Running command echo 'TMOUT=600' >> ~/.bashrc on x.x.x.x
20:36:41 - MainThread - ocs_ci.ocs.external_ceph - INFO - Command completed successfully
20:36:41 - MainThread - ocs_ci.ocs.external_ceph - INFO - Running command [ -f /etc/redhat-release ] on x.x.x.x
20:36:42 - MainThread - ocs_ci.ocs.external_ceph - INFO - finished connect

**Comment by shylesh** [comment]
Verified locally

File: ocs_ci/ocs/utils.py [hunk 1]
```diff
@@ -1012,3 +1013,24 @@ def revive_osd_external(ceph_cluster, osd_id):
...
             raise
+
+
+def reboot_node(ceph_node, timeout=300):
+    """
+    Reboot a node with given ceph_node object
+
+    Args:
+        ceph_node (CephNode): ceph node object representing the node
+        timeout (int): Wait time for node to comeback
```
Comment (petr-balogh) [inline ocs_ci/ocs/utils.py:1024]:
```suggestion
        timeout (int): Wait time in seconds for node to come back.
```

File: ocs_ci/ocs/utils.py [hunk 2]
```diff
@@ -1012,3 +1013,24 @@ def revive_osd_external(ceph_cluster, osd_id):
...
             log.error("Failed to revive osd")
             raise
+
+
+def reboot_node(ceph_node, timeout=300):
+    """
+    Reboot a node with given ceph_node object
+
+    Args:
+        ceph_node (CephNode): ceph node object representing the node
```
Comment (petr-balogh) [inline ocs_ci/ocs/utils.py:1023]:
```suggestion
        ceph_node (CephNode): Ceph node object representing the node.
```
nitpick but try to be consistent with docstring and if you start one with Capital letter, follow it on all descriptions. And if I am starting with Capital I would also add period at the end of sentence.

File: ocs_ci/ocs/utils.py [hunk 3]
```diff
@@ -1012,3 +1013,24 @@ def revive_osd_external(ceph_cluster, osd_id):
...
+        timeout (int): Wait time for node to comeback
+
+    """
+    ceph_node.exec_command(
+        cmd='reboot',
+        check_ec=False,
+        long_running=True
```
Comment (petr-balogh) [inline ocs_ci/ocs/utils.py:1030]:
```suggestion
        long_running=True,
```

Black will add it as well I think so maybe you can do it in this PR directly ;)

File: ocs_ci/ocs/utils.py [hunk 4]
```diff
@@ -1012,3 +1013,24 @@ def revive_osd_external(ceph_cluster, osd_id):
...
+    )
+
+    try:
+        ceph_node.connect(timeout)
+    except SSHException:
+        log.exception(f"Failed to connect to node {ceph_node.hostname}")
```
Comment (petr-balogh) [inline ocs_ci/ocs/utils.py:1036]: Don't we want to re-raise the exception
```suggestion
        log.exception(f"Failed to connect to node {ceph_node.hostname}")
        raise
```

Or raise some specific exception which we will define in the docstring? No need for extra exception and we can document maybe directly SSHExceptin if we really want to raise an exception. But I think that function should raise an exception or return boolean based on success/failure.

**Review by vasukulkarni — APPROVED** [review]

File: ocs_ci/ocs/utils.py (hunk 4 above)
Comment (shylesh) [reply ocs_ci/ocs/utils.py:1038]: done

**Review by clacroix12 — APPROVED** [review]

**Review by petr-balogh — APPROVED** [review]

**Pull request merged**